### PR TITLE
PR: Editor historystack improvements

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes: #5226 

I think the error was the unnecessary side effect of `_get_previous_file_index`, although I wasn't able to reproduce #5226

### CODE REVIEW

File: spyder/widgets/editor.py [hunk 1]
```diff
@@ -304,7 +304,12 @@ def __len__(self):
         return len(self.history)
 
     def __getitem__(self, i):
-        return self.id_list.index(self.history[i])
+        self._update_id_list()
+        try:
+            return self.id_list.index(self.history[i])
+        except ValueError:
+            self.refresh
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: This should be `self.refresh()`, right?

File: spyder/widgets/editor.py [hunk 2]
```diff
@@ -304,7 +304,12 @@ def __len__(self):
         return len(self.history)
 
     def __getitem__(self, i):
-        return self.id_list.index(self.history[i])
+        self._update_id_list()
+        try:
+            return self.id_list.index(self.history[i])
+        except ValueError:
+            self.refresh
+            raise IndexError
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: Do we really need to raise an `IndexError` here?

File: spyder/widgets/editor.py [hunk 3]
```diff
@@ -1718,12 +1723,11 @@ def current_changed(self, index):
                 pass
 
     def _get_previous_file_index(self):
-        if len(self.stack_history) > 1:
-            last = len(self.stack_history)-1
-            w_id = self.stack_history.pop(last)
-            self.stack_history.insert(0, w_id)
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: Why did you remove this?

File: spyder/widgets/editor.py (hunk 1 above)
Comment (rlaverde) [reply spyder/widgets/editor.py]: Yes, thanks for noticing it

File: spyder/widgets/editor.py (hunk 2 above)
Comment (rlaverde) [reply spyder/widgets/editor.py]: I think it's the expected bevahior, when an elementary isn't found in a list an IndexError is raised.
The other option will be to return None.

File: spyder/widgets/editor.py (hunk 3 above)
Comment (rlaverde) [reply spyder/widgets/editor.py]: This was an unnecessary side effect, that could cause the corruption of the historystack order.

This reorder the editor stack before closing the file, the closing could be cancelled, an the editor stack will have a wrong order.

Also this was unnecessary, because when a file is closed the CurrentChanged  signal is be emited, and that cause the history_stack to be updated.

Comment (ccordoba12) [reply spyder/widgets/editor.py]: Ok, thanks for the explanation :-)
